shell/browser/notifications/win/windows_toast_notification.cc

```diff
@@ -265,13 +265,136 @@ HRESULT WindowsToastNotification::SetXmlScenarioReminder(IXmlDocument* doc) {
+  ComPtr<IXmlNode> action_node;
+  RETURN_IF_FAILED(
+      actions_wrapper_node->AppendChild(action_node_tmp.Get(), &action_node));
+
+  // Setup attributes for action
+  ComPtr<IXmlNamedNodeMap> action_attributes;
+  RETURN_IF_FAILED(action_node->get_Attributes(&action_attributes));
+
+  // Create activationType attribute
+  ComPtr<IXmlAttribute> activation_type_attribute;
+  ScopedHString activation_type_str(L"activationType");
+  RETURN_IF_FAILED(
+      doc->CreateAttribute(activation_type_str, &activation_type_attribute));
+
+  ComPtr<IXmlNode> activation_type_attribute_node;
+  RETURN_IF_FAILED(
+      activation_type_attribute.As(&activation_type_attribute_node));
+
+  // Set activationType attribute to system
+  ScopedHString activation_type_value(L"system");
+  if (!activation_type_value.success())
+    return E_FAIL;
+
+  ComPtr<IXmlText> activation_type_text;
+  RETURN_IF_FAILED(
+      doc->CreateTextNode(activation_type_value, &activation_type_text));
+
+  ComPtr<IXmlNode> activation_type_node;
+  RETURN_IF_FAILED(activation_type_text.As(&activation_type_node));
+
+  ComPtr<IXmlNode> activation_type_backup_node;
+  RETURN_IF_FAILED(activation_type_attribute_node->AppendChild(
+      activation_type_node.Get(), &activation_type_backup_node));
+
+  // Add activation type to the action attributes
+  ComPtr<IXmlNode> activation_type_attribute_pnode;
+  RETURN_IF_FAILED(action_attributes.Get()->SetNamedItem(
+      activation_type_attribute_node.Get(), &activation_type_attribute_pnode));
+
+  // Create arguments attribute
+  ComPtr<IXmlAttribute> arguments_attribute;
+  ScopedHString arguments_str(L"arguments");
+  RETURN_IF_FAILED(doc->CreateAttribute(arguments_str, &arguments_attribute));
+
+  ComPtr<IXmlNode> arguments_attribute_node;
+  RETURN_IF_FAILED(arguments_attribute.As(&arguments_attribute_node));
+
+  // Set arguments attribute to dismiss
+  ScopedHString arguments_value(L"dismiss");
+  if (!arguments_value.success())
+    return E_FAIL;
+
+  ComPtr<IXmlText> arguments_text;
+  RETURN_IF_FAILED(doc->CreateTextNode(arguments_value, &arguments_text));
+
+  ComPtr<IXmlNode> arguments_node;
+  RETURN_IF_FAILED(arguments_text.As(&arguments_node));
+
+  ComPtr<IXmlNode> arguments_backup_node;
+  RETURN_IF_FAILED(arguments_attribute_node->AppendChild(
+      arguments_node.Get(), &arguments_backup_node));
+
+  // Add arguments to the action attributes
+  ComPtr<IXmlNode> arguments_attribute_pnode;
+  RETURN_IF_FAILED(action_attributes.Get()->SetNamedItem(
+      arguments_attribute_node.Get(), &arguments_attribute_pnode));
+
+  // Create content attribute
+  ComPtr<IXmlAttribute> content_attribute;
+  ScopedHString content_str(L"content");
+  RETURN_IF_FAILED(doc->CreateAttribute(content_str, &content_attribute));
+
+  ComPtr<IXmlNode> content_attribute_node;
+  RETURN_IF_FAILED(content_attribute.As(&content_attribute_node));
+
+  // Set content attribute to Dismiss
+  ScopedHString content_value(L"Dismiss");
```

Comment:
At the moment this is hardcoded to `Dismiss` - ideally we'll want to ensure that this gets localized to the user's current location. We can do this with the localization utility from Chromium:

```
#include "ui/base/l10n/l10n_util.h"
...
stuff
...
// Set content attribute to Dismiss
ScopedHString content_value(l10n_util::GetStringUTF16(IDS_TOAST_NOTIFICATION_DISMISS_TEXT));
```

We'll also want to add this string to our list of generated strings in `electron_strings.grdp` i would say, something like:

```
<message name="IDS_TOAST_NOTIFICATION_DISMISS_TEXT" desc="Text label for the Dismiss button in Windows Toast Notifications. This is used when the toast has a timeout type of never.">
      Dismiss
</message>
```

in the [windows-only block](https://github.com/electron/electron/blob/8bbdc224ac3d7756af930ce946a59ce5ff2533b5/electron_strings.grdp#L78).

Reply:
Done!

Reply:
Chromium has the same issue and has similar code to inject a button to prevent the notification from timing out -- can we piggyback on their strings, e.g. `IDS_APP_CLOSE` instead of creating another one?